crates/amaru-consensus/src/consensus/stages/receive_header.rs

```diff
@@ -46,14 +46,33 @@ pub async fn stage(
         } => {
             let header = match decode_header(&point, raw_header.as_slice()) {
                 Ok(header) => {
-                    let result = eff.store().store_header(&header);
-                    if let Err(error) = result {
+                    if header.point() != point {
+                        tracing::error!(%point, %peer, "Header point {} does not match expected point", header.point());
                         eff.base()
-                            .send(&errors, ProcessingFailed::new(&peer, anyhow!(error)))
+                            .send(
+                                &failures,
+                                ValidationFailed::new(
+                                    &peer,
+                                    ConsensusError::CannotDecodeHeader {
```

Comment:
I would rather have an explicit variant for this specific error, rather than having to unpack the `reason` field to know exactly why the validation failed.